### DISCUSSION TITLE
12. AOP에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by KIMSEI1124** [Study-Diary/Computer-Science#49] [Spring]
- @Aspect는 어떻게 동작하나요?

**Comment by jiwon83** [comment]
AOP를 사용하는 대신 Util 클래스를 만들어서 생성하는 것이 낫지 않는가?

**Reply by KIMSEI1124** [reply]
`Around`와 같은 경우는 유틸클래스를 사용한다면 메서드를 두번 호출해야하고 만약, 타이머와 같은 경우라면 인스턴스 값을 가지고 있어야 하므로 이러한 경우에는 AOP를 사용하는게 좋을 것 같습니다.

**Comment by jiwon83** [comment]
프록시의 사용했을 떄의 장점은 무엇일까요?

**Comment by jiwon83** [comment]
프록시가 메서드 오버라이딩 개념으로 동작한다고 했는데, 구체적으로 어떠한 작동 과정으로 동작하는 지 궁금합니다.
